### PR TITLE
Fix parsing of SWC files with discontinuous section points.

### LINKED ISSUE #383: Inconsistencies between ezy and fst
Example of disordered data [test_data.zip](https://github.com/BlueBrain/NeuroM/files/318302/test_data.zip). The two files should be equivalent (the only difference is the order in which the points appear in the file):

```python
from neurom import ezy
from neurom import fst

n = ezy.load_neuron('sample.swc')
n1 = ezy.load_neuron('sample_disordered.swc')

ezy.get('number_of_sections_per_neurite', n)
Out: array([ 5.,  3.])

ezy.get('number_of_sections_per_neurite', n1)
Out: array([ 5.,  3.])

n0 = fst.load_neuron('sample.swc')
n01 = fst.load_neuron('sample_disordered.swc')

Out:
KeyError                                  Traceback (most recent call last)
/home/kanari/bitbucket/NeuronTopology/code/<ipython-input-7-567d250915ee> in <module>()
----> 1 n01 = fst.load_neuron('./neurontopology/test_data/sample_disordered.swc')

/home/kanari/github/Experimental_ready/NeuroM/neurom/fst/_io.pyc in load_neuron(filename)
    113     }
    114     ext = os.path.splitext(filename)[1][1:]
--> 115     rdw = _READERS[ext.lower()](filename)
    116     trees = make_trees(rdw, _NEURITE_ACTION[ext.lower()])
    117     soma = make_soma(rdw.soma_points())

/home/kanari/github/Experimental_ready/NeuroM/neurom/fst/_io.pyc in <lambda>(f)
    103     '''Build section trees from an h5 or swc file'''
    104     _READERS = {
--> 105         'swc': lambda f: SWC.read(f, wrapper=SecDataWrapper),
    106         'h5': lambda f: H5.read(f, remove_duplicates=False, wrapper=SecDataWrapper),
    107         'asc': lambda f: NeurolucidaASC.read(f, remove_duplicates=False, wrapper=SecDataWrapper)

/home/kanari/github/Experimental_ready/NeuroM/neurom/io/swc.pyc in read(filename, wrapper)
    110             data = np.reshape(data, (1, -1))
    111         data = data[:, [SWC.X, SWC.Y, SWC.Z, SWC.R, SWC.TYPE, SWC.ID, SWC.P]]
--> 112         return wrapper(data, 'SWC', None)

/home/kanari/github/Experimental_ready/NeuroM/neurom/fst/_io.pyc in __init__(self, data_block, fmt, sections)
     54         self.data_block = data_block
     55         self.fmt = fmt
---> 56         self.sections = sections if sections is not None else extract_sections(data_block)
     57 
     58     def neurite_trunks(self):

/home/kanari/github/Experimental_ready/NeuroM/neurom/fst/_io.pyc in extract_sections(data_block)
    170     for sec in _sections:
    171         if sec.ids:
--> 172             sec.pid = parent_section[sec.ids[0]]
    173 
    174     return [s for s in _sections if s.ids]

KeyError: 31

fst.get('number_of_sections_per_neurite', n0)
Out: array([5, 3])
```

The disordered file fails to load with the fst module!

### PR DESCRIPTION
**Pull request by juanchopanza**
Fixes #383.

### CODE REVIEW

File: neurom/fst/_io.py [hunk 1]
```diff
@@ -111,6 +111,30 @@ def load_neuron(filename):
...
+    n_children = defaultdict(int)
+    for row in data_block:
+        n_children[int(row[COLS.P])] += 1
+
+    # end points have either no children or more than one
+    return set(i for i, row in enumerate(data_block)
+               if n_children[row[COLS.ID]] != 1)
+
+
 def extract_sections(data_block):
```
Comment (juanchopanza) [inline neurom/fst/_io.py]: This function is huge and awful. I need to find a better way to do this in the future.

**Comment by juanchopanza** [comment]
@lidakanari I think this fixes the issue (albeit in an awful way.)

File: neurom/fst/tests/test_io.py [hunk 2]
```diff
@@ -89,3 +90,22 @@ def test_load_neurons_filenames():
...
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix),
+                          get('number_of_sections_per_neurite', SWC_ORD_REF))
+
+
+def test_load_neuron_section_order_break_swc():
+    nrn_mix =  _io.load_neuron(os.path.join(SWC_PATH, 'sample_disordered.swc'))
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix), [5, 3])
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix),
+                          get('number_of_sections_per_neurite', SWC_ORD_REF))
+
```
Comment (lidakanari) [inline neurom/fst/tests/test_io.py]: Is it possible to add another test to ensure that the rest of the data are correct? Something along the following lines:
```python
get('number_of_segments') == REF
get('total_length') == REF
```

Comment (juanchopanza) [inline neurom/fst/tests/test_io.py]: I will, but I will have to change the "mixed" file. I swapped two trunk sections, and that affects the total length.

Comment (lidakanari) [inline neurom/fst/tests/test_io.py]: OK, no problem! I just think it would be safer if we have more accurate tests...

**Pull request merged**